Library/Homebrew/cask/lib/hbc/caskroom.rb

```diff
@@ -13,32 +13,32 @@ def migrate_caskroom_from_repo_to_prefix
         FileUtils.mv repo_caskroom, Hbc.caskroom
       else
         opoo "#{Hbc.caskroom.parent} is not writable, sudo is needed to move the Caskroom."
-        system "/usr/bin/sudo", "--", "/bin/mv", "--", repo_caskroom.to_s, Hbc.caskroom.parent.to_s
+        command "/bin/mv", repo_caskroom, Hbc.caskroom.parent, :use_sudo => true
       end
     end
 
     def ensure_caskroom_exists
       return if Hbc.caskroom.exist?
 
       ohai "Creating Caskroom at #{Hbc.caskroom}"
-      if Hbc.caskroom.parent.writable?
-        Hbc.caskroom.mkpath
-      else
-        ohai "We'll set permissions properly so we won't need sudo in the future"
-        toplevel_dir = Hbc.caskroom
-        toplevel_dir = toplevel_dir.parent until toplevel_dir.parent.root?
-        unless toplevel_dir.directory?
-          # If a toplevel dir such as '/opt' must be created, enforce standard permissions.
-          # sudo in system is rude.
-          system "/usr/bin/sudo", "--", "/bin/mkdir", "--",         toplevel_dir
-          system "/usr/bin/sudo", "--", "/bin/chmod", "--", "0775", toplevel_dir
-        end
-        # sudo in system is rude.
-        system "/usr/bin/sudo", "--", "/bin/mkdir", "-p", "--", Hbc.caskroom
-        unless Hbc.caskroom.parent == toplevel_dir
-          system "/usr/bin/sudo", "--", "/usr/sbin/chown", "-R", "--", "#{Utils.current_user}:staff", Hbc.caskroom.parent.to_s
-        end
+      ohai "We'll set permissions properly so we won't need sudo in the future"
+      use_sudo = !Hbc.caskroom.parent.writable?
+
+      command "/bin/mkdir", "-p", Hbc.caskroom, :use_sudo => use_sudo
+      command "/bin/chmod", "g+rwx", Hbc.caskroom, :use_sudo => use_sudo
+      command "/usr/sbin/chown", Utils.current_user, Hbc.caskroom, :use_sudo => use_sudo
+      command "/usr/bin/chgrp", "admin", Hbc.caskroom, :use_sudo => use_sudo
+    end
+
+    def command(*args)
```

Comment:
Use `SystemCommand` instead of defining another wrapper for `system`.

```ruby
SystemCommand.run("/bin/mkdir", args: ["-p", Hbc.caskroom], sudo: use_sudo)
```